internal/provider/resource_value.go

```diff
@@ -91,10 +112,40 @@ func (r *ResourceValue) Schema(ctx context.Context, req resource.SchemaRequest,
 				},
 			},
 			"value": schema.StringAttribute{
-				MarkdownDescription: "The value that will be stored.",
-				Required:            true,
+				MarkdownDescription: "The value that will be stored. It can't be defined if secret_ref is defined.",
+				Optional:            true,
 				Sensitive:           true,
 			},
+			"secret_ref": schema.SingleNestedAttribute{
+				MarkdownDescription: "The sensitive value that will be stored in the primary organization store or a reference to a sensitive value already stored in one of the registered stores. It can't be defined if is_secret is false or value is defined.",
+				Optional:            true,
+				Computed:            true,
+				Attributes: map[string]schema.Attribute{
+					"ref": schema.StringAttribute{
+						MarkdownDescription: "Secret reference in the format of the target store. It can't be defined if value is defined.",
+						Optional:            true,
+						Computed:            true,
+						PlanModifiers:       []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
```

Comment:
These seem needed to avoid error I saw in CI when `secret_ref.value` was set (e.g. [here](https://github.com/humanitec/terraform-provider-humanitec/actions/runs/6488404943/job/17620772725#step:5:101)).